pallets/cash/src/internal/notices.rs

```diff
@@ -89,17 +95,11 @@ pub fn publish_signature(
                 _ => Err(Reason::SignatureMismatch),
             }?;
 
-            // require(signer == known validator); // XXX
-
-            // XXX sets?
-            // let signers_new = {signer | signers};
-            // let signatures_new = {signature | signatures };
+            // Note: we currently iterate all potentially all validators to check validity
+            if !Validators::iter().any(|(_,v)| ChainAccount::Eth(v.eth_address) == signer) {
```

Comment:
so we do the sigs first and then check if the signer is a validator? kid of weird but :shrug

Reply:
good question on ordering, but I think either way works. We can optimize between the two.